src/System.Memory/src/System/SpanExtensions.cs

```diff
@@ -32,6 +31,10 @@ public static int IndexOf<T>(this Span<T> span, T value)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static int IndexOf(this Span<byte> span, byte value)
         {
+            if (!BitConverter.IsLittleEndian)
```

Comment:
It may suffer from a static base initalization check if its first use; and not do branch elimination. Subsequent uses (in different function) will branch eliminate.

@jkotas would there me any mileage in adding a use of `BitConverter.IsLittleEndian` and `IntPtr.Size` to something like [`CommonlyUsedGenericInstantiations`](https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/Internal.cs#L47) as they are runtime fixed and mainly used for branch elimination; so the check is pre-paid?